src/app.md

```diff
@@ -725,6 +726,119 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given a list of version &strs, finds the latest [semver::Version] that satisfying given [semver::VersionReq] "<= 1.0.0" using [VersionReq::matches].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::{Version, VersionReq, AlphaNumeric, Numeric};
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::ReqParseError);
+#     }
+# }
+
+fn version_latest<'a>(
+    version_req_str: &str,
+    version_str_vec: &'a Vec<&str>,
+) -> Result<Version, &'a str> {
+    let vreq = VersionReq::parse(version_req_str)?;
+
+    version_str_vec
+        .iter()
+        .filter_map(|s| Version::parse(s).ok())
+        .filter(|s| vreq.matches(s))
+        .max()
+        .ok_or_else(|| "Unable to find version satisfying given range")
+}
+
+fn run() -> Result<()> {
+/*
+    let req = VersionReq::parse("<= 1.0.0")?;
+
+    let ver_latest = vec!["0.9.0", "1.0.0", "1.0.1"]
+        .iter()
+        .filter_map(|s| Version::parse(s).ok())
+        .filter(|s| req.matches(s))
+        .max()
+        .ok_or_else(|| "Unable to find version satisfying given range");
+
+    assert_eq!(
+        ver_latest,
+        Ok(Version {
+            major: 1,
+            minor: 0,
+            patch: 0,
+            pre: vec![],
+            build: vec![],
+        })
+    );
+
+    let req = VersionReq::parse(">1.2.3-alpha.3")?;
+
+    let ver_latest = vec!["1.2.3-alpha.3", "1.2.3-alpha.4", "1.2.3-alpha.10", "3.4.5-alpha.9"]
+        .iter()
+        .filter_map(|s| Version::parse(s).ok())
+        .filter(|s| req.matches(s))
+        .max()
+        .ok_or_else(|| "Unable to find version satisfying given range");
+
+    assert_eq!(
+        ver_latest,
+        Ok(Version {
+            major: 1,
+            minor: 2,
+            patch: 3,
+            pre: vec![AlphaNumeric(String::from("alpha")), Numeric(10)],
+            build: vec![],
+        })
+    );
+*/
+
+    assert_eq!(
+        version_latest("<= 1.0.0", &vec!["0.9.0", "1.0.0", "1.0.1"]),
+        Ok(Version {
```

Comment:
error-chain generated `Result`'s tend not to have `PartialEq` implemented so such assertion will not compile but one in which you'd use `?` after the function call will.